src/main/groovy/com/github/winplay02/gitcraft/mappings/MappingFlavour.java

```diff
@@ -1,28 +1,108 @@
 package com.github.winplay02.gitcraft.mappings;
 
-import com.github.winplay02.gitcraft.GitCraft;
-
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
 import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
 
-public enum MappingFlavour {
-	MOJMAP(GitCraft.MOJANG_MAPPINGS),
-	FABRIC_INTERMEDIARY(GitCraft.FABRIC_INTERMEDIARY_MAPPINGS),
-	YARN(GitCraft.YARN_MAPPINGS),
-	MOJMAP_PARCHMENT(GitCraft.MOJANG_PARCHMENT_MAPPINGS);
+import com.github.winplay02.gitcraft.mappings.ornithe.CalamusIntermediaryMappings;
+import com.github.winplay02.gitcraft.mappings.ornithe.FeatherMappings;
+import com.github.winplay02.gitcraft.mappings.yarn.FabricIntermediaryMappings;
+import com.github.winplay02.gitcraft.mappings.yarn.YarnMappings;
+import com.github.winplay02.gitcraft.pipeline.StepStatus;
+import com.github.winplay02.gitcraft.pipeline.StepWorker;
+import com.github.winplay02.gitcraft.pipeline.key.MinecraftJar;
+import com.github.winplay02.gitcraft.types.OrderedVersion;
+import com.github.winplay02.gitcraft.util.LazyValue;
 
-	private final Mapping mappingImpl;
+import net.fabricmc.mappingio.MappingVisitor;
+import net.fabricmc.tinyremapper.IMappingProvider;
 
-	MappingFlavour(Mapping mapping) {
-		this.mappingImpl = mapping;
-	}
+public enum MappingFlavour {
+	MOJMAP(MojangMappings::new),
+	FABRIC_INTERMEDIARY(FabricIntermediaryMappings::new),
+	YARN(() -> new YarnMappings((FabricIntermediaryMappings) FABRIC_INTERMEDIARY.impl.get())),
+	MOJMAP_PARCHMENT(() -> new ParchmentMappings((MojangMappings) MOJMAP.impl.get())),
+	CALAMUS_INTERMEDIARY(CalamusIntermediaryMappings::new),
+	FEATHER(FeatherMappings::new),
+	IDENTITY_UNMAPPED(IdentityMappings::new),
+	MOJMAP_YARN(() -> new MojangPlusYarnMappings((MojangMappings) MOJMAP.impl.get(), (YarnMappings) YARN.impl.get()));
```

Comment:
similar to above, I wonder if this can be generalized to a `CombinedMapping` - may be for a separate PR though